src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ModelExplorerExtensions.cs

```diff
@@ -40,6 +41,25 @@ public static string GetSimpleDisplayText(this ModelExplorer modelExplorer)
                 return modelExplorer.Metadata.NullDisplayText;
             }
 
+            if (modelExplorer.Metadata.IsEnum)
+            {
+                var modelEnum = (Enum) modelExplorer.Model;
+                var enumStringValue = modelEnum.ToString("d");
+                var enumGroupedDisplayNamesAndValues = modelExplorer.Metadata.EnumGroupedDisplayNamesAndValues;
+
+                Debug.Assert(enumGroupedDisplayNamesAndValues != null);
+
+                var enumDisplayValue = enumGroupedDisplayNamesAndValues
+                    .Where(enumGroup => enumGroup.Value == enumStringValue)
+                    .Select(kv => kv.Key.Name)
+                    .FirstOrDefault(value => value != null);
```

Comment:
Avoid Linq here, use `sting.Equals(...)`, and should be no need to `null`-check the `Name` e.g.
``` c#
string enumDisplayValue = null;
foreach (var kvp in enumGroupedDisplayNamesAndValues)
{
    if (string.Equals(kvp.Value, enumStringValue, StringComparison.Ordinal))
    {
        enumStringValue = kvp.Key.Name;
        break;
    }
}
```

Reply:
Out of curiosity, I'm guessing string.Equals is preferred because it gives you the ordinal option?